lib/index.js

```diff
@@ -53,13 +53,40 @@ function applyMethods(q) {
     });
   };
 
-  q.series = function(arr, body) {
+  q.series = function (arr, body) {
     return arr.reduce(function(promise, item) {
       return promise.then(function(result) {
         return body(item, result);
       });
     }, q());
   };
+
+  /**
+   * Allows body up to timeout ms to complete, and retries it if exceeds that time.
+   * Body will not be retried if it fails for any other reason (e.g. exception or rejection).
+   */
+  q.timeoutRetry = function (timeout, body, options) {
+    options = options || {};
+
+    var timeoutIdentifier = 'timeout-dc1836808a';
```

Comment:
Per offline remark, leaving as `var`, so that it can work in older JS (browsers?) should the need arise.